lib/prompts/createApiSamplePrompt.ts

```diff
@@ -1,63 +1,90 @@
+import { i18n } from '../lang';
+import { PromptOperand, PromptConfig } from '../../types/prompts';
 
 const i18nKey = 'lib.prompts.createApiSamplePrompt';
 
-const getSampleTypesPrompt = choices => ({
-  type: 'rawlist',
-  name: 'sampleType',
-  message: i18n(`${i18nKey}.selectApiSampleApp`),
-  choices: choices.map(choice => ({
-    name: `${choice.name} - ${choice.description}`,
-    value: choice.id,
-  })),
-  validate: input => {
-    return new Promise(function(resolve, reject) {
-      if (input.length > 0) {
-        resolve(true);
-      }
-      reject(i18n(`${i18nKey}.errors.apiSampleAppRequired`));
-    });
-  },
-});
+type SampleChoice = {
+  name: string;
+  description: string;
+  id: string;
+  languages: string[];
+};
+
+type SampleConfig = {
+  samples: SampleChoice[];
+};
+
+type CreateApiSamplePromptResponse = {
+  sampleType?: string;
+  sampleLanguage?: string;
+};
+
+function getSampleTypesPrompt(
+  choices: SampleChoice[]
+): PromptConfig<CreateApiSamplePromptResponse> {
+  return {
+    type: 'rawlist',
+    name: 'sampleType',
+    message: i18n(`${i18nKey}.selectApiSampleApp`),
+    choices: choices.map(choice => ({
+      name: `${choice.name} - ${choice.description}`,
+      value: choice.id,
+    })),
+    validate: function(input?: string) {
+      return new Promise<PromptOperand>(function(resolve, reject) {
+        if (input && input.length > 0) {
+          resolve(true);
+        } else {
+          reject(i18n(`${i18nKey}.errors.apiSampleAppRequired`));
+        }
+      });
+    },
+  };
+}
 
-const getLanguagesPrompt = choices => ({
-  type: 'rawlist',
-  name: 'sampleLanguage',
-  message: i18n(`${i18nKey}.selectLanguage`),
-  choices: choices.map(choice => ({
-    name: choice,
-    value: choice,
-  })),
-  validate: input => {
-    return new Promise(function(resolve, reject) {
-      if (input.length > 0) {
-        resolve(true);
-      }
-      reject(i18n(`${i18nKey}.errors.languageRequired`));
-    });
-  },
-});
+function getLanguagesPrompt(
+  choices: string[]
+): PromptConfig<CreateApiSamplePromptResponse> {
+  return {
+    type: 'rawlist',
+    name: 'sampleLanguage',
+    message: i18n(`${i18nKey}.selectLanguage`),
+    choices: choices.map(choice => ({
+      name: choice,
+      value: choice,
+    })),
+    validate: function(input: string | undefined) {
+      return new Promise<PromptOperand>(function(resolve, reject) {
+        if (input && input.length > 0) {
+          resolve(true);
+        }
+        reject(i18n(`${i18nKey}.errors.languageRequired`));
+      });
+    },
+  };
+}
 
-const createApiSamplePrompt = async samplesConfig => {
+export async function createApiSamplePrompt(
+  samplesConfig: SampleConfig
+): Promise<CreateApiSamplePromptResponse> {
   try {
     const { samples } = samplesConfig;
-    const sampleTypeAnswer = await promptUser(getSampleTypesPrompt(samples));
+    const sampleTypeAnswer = await promptUser<CreateApiSamplePromptResponse>(
```

Comment:
This is maybe a little pedantic, but the typing here isn't entirely accurate. `createApiSamplePrompt` does return a `CreateApiSamplePromptResponse`, but the individual `promptUser`s in this function don't - they're only returning one field that we know exists (`sampleType` and `sampleLanguage` respectively). Might be a little clearer to give them each their own type

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1288/commits/d73f2185f196a582671543443e00908845022b7c